source/rules/expressions/primaries.rs

```diff
@@ -184,6 +200,71 @@ fn empty_mapper<'a>(expression: Expression<'a>) -> StdResult<Expression<'a>, ()>
     Ok(Expression::Empty(Box::new(expression)))
 }
 
+named!(
+    intrinsic_eval<Expression>,
+    map_res!(
+        preceded!(
+            keyword!(tokens::EVAL),
+            preceded!(
+                first!(tag!(tokens::LEFT_PARENTHESIS)),
+                terminated!(
+                    first!(expression),
+                    first!(tag!(tokens::RIGHT_PARENTHESIS))
+                )
+            )
+        ),
+        eval_mapper
+    )
+);
+
+#[inline(always)]
+fn eval_mapper<'a>(expression: Expression<'a>) -> StdResult<Expression<'a>, ()> {
+    Ok(Expression::Eval(Box::new(expression)))
+}
+
+named!(
+    intrinsic_exit<Expression>,
+    map_res!(
+        preceded!(
+            alt!(
+                keyword!(tokens::EXIT)
+              | keyword!(tokens::DIE)
+            ),
+            opt!(
+                preceded!(
+                    first!(tag!(tokens::LEFT_PARENTHESIS)),
+                    terminated!(
+                        first!(expression),
+                        first!(tag!(tokens::RIGHT_PARENTHESIS))
+                    )
+                )
+            )
+        ),
+        exit_mapper
+    )
+);
+
+#[inline(always)]
+fn exit_mapper<'a>(expression: Option<Expression<'a>>) -> StdResult<Expression<'a>, Error<ErrorKind>> {
+    match expression {
+        Some(expression) => {
+            if let Expression::Literal(Literal::Integer(code)) = expression {
+                if code == 255 {
+                    return Err(Error::Code(ErrorKind::Custom(IntrinsicError::ReservedExitCode as u32)));
+                } else if code > 255 {
+                    return Err(Error::Code(ErrorKind::Custom(IntrinsicError::OutOfRangeExitCode as u32)));
+                }
```

Comment:
I feel like if this can checked at parse-time easily, we should do it to avoid any “hidden error” (at least a test case should cover this instruction at runtime). Of course, `$a = 12; exit($a);` could be checked at runtime (or with a static or concolic approach for instance), but if we can avoid a malformed instruction easily, I feel we should do it. Thoughts?